Fix #22560: blog link needed at drop down menu

TejasSaraf opened the pull request:
## Overview

1. This PR fixes or fixes part of #22560 .
2. This PR does the following: [Added the blog link at the drop-down of About menu]

## Essential Checklist

- [x] I have linked the issue that this PR fixes in the "Development" section of the sidebar.
- [x] I have checked the "Files Changed" tab and confirmed that the changes are what I want to make.
- [x] I have written tests for my code.
- [x] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [x] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I can't assign them directly).


## Proof that changes are correct

Before 

https://github.com/user-attachments/assets/e36a46c8-0f47-4ce2-b2aa-27731526cbc3

After

https://github.com/user-attachments/assets/8e187b80-d392-46b6-8511-062a709fb3db


#### Proof of changes in Arabic language

![image](https://github.com/user-attachments/assets/ebf0a3a4-987f-46dc-b2e6-cea2d9ce2755)

TejasSaraf committed:
Added blog page link in the drop-down of about menu

TejasSaraf committed:
Fixed errors

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

Nik-09 commented:
Hi @TejasSaraf few tests are failing, and once the PR is ready for reviews, please follow step no. 5 in the "Essential checklist" section of the PR description to assign reviewers.
Thanks!

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

TejasSaraf committed:
Fixed test errors

TejasSaraf committed:
Added navigation tests

TejasSaraf committed:
Fixing lexicographic order

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

TejasSaraf committed:
Added blog link to mobile side navigation bar

TejasSaraf committed:
Fixed indentation

TejasSaraf committed:
Fixed lexicographic order

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

TejasSaraf commented:
Hello @Nik-09, all checks have passed PTAL!

charliemax5 committed:
Fix Issue #20463: Adds Warning Message for Invalid Activity IDs entered on Moderator Page Featured Activities Tab (#22453)

* Changed require_activities_to_be_public() function to check_activity_id_validity() to return lists of IDs for each kind of invalid activity

* Edited moderator.py to retrieve lists of invalid activities and customize an error message when there is a POST request

* Edited moderator-page.component.ts to handle error from backend and display the custom message with alertsService

* Edited moderator-page.component.spec.ts tests to reflect new functionality and added new tests for each invalid ID scenario

* Edited summary_services_test.py to better reflect new functionality of check_activity_id_validity()

* Edited existing tests and created new tests in moderator_test.py to test for deveral possible scenarios of Invalid IDs

* Fixing Linting Errors in moderator_test.py

* Fixing Linting Errors in moderator_test.py

* Fixing Linting Errors in moderator_test.py

* Fixing Errors in moderator_test.py

* Trying different way to organize tests in moderator_test.py

* Edited code in accordance with review

* Fixing Linter errors

* Changed variable names and reconfigured code according to review comments

* Changed Variable names and Test names per review request

* Linting errors

GusGus17 committed:
Fix #22109: Topic manager doesn't get any error messages editing the topic name anymore (#22346)

* Topic manager doesn't get any error messages editing the topic name anymore

Files changed fixing the bug:
-topic_editor.py: Removed the TopicNameHandler class
-topic_viewer.py: Added the TopicNameHandler class with a open_access decorator
 and added an import statement to be able to use it.
-main.py: Changed the URL routing to point to the new location
 of topicNameHandler in topic_viewer.py

Files changed for testing the fix:
-topic_viewer_test.py: Added class TopicNameHandlerTests with 3 tests.
The first two, "test_get_with_existing_topic_name" and
"test_get_with_nonexistent_topic_name" test if the handler
 correctly identifies existing and non-existing topic names.
And finally "test_get_with_guest_user" tests if the handler
is accessible to all users since it's using an open_access decorator

* Fix tests

* Fix testhandlers

---------

Co-authored-by: Hardik Goyal <135348086+HardikGoyal2003@users.noreply.github.com>

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/logged-out-user.ts:
```diff
@@ -808,6 +811,31 @@ export class LoggedOutUser extends BaseUser {
+   * Function to click the Blog button in the About Menu on navbar
+   * and check if it opens the Blog page.
+   */
+  async clickBlogButtonInAboutMenuOnNavbar(): Promise<void> {
```
Why is this function added? I don't see this function being used in any files.

TejasSaraf replied:
You are right I made the necessary changes.

imchristie reviewed (commented):
Left one comment.

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

TejasSaraf committed:
Corrected Translation

TejasSaraf committed:
Removed unused function

TejasSaraf committed:
Removed unused code

Nik-09 commented on assets/i18n/es.json:
```diff
@@ -1743,6 +1743,7 @@
 	"I18N_TOPNAV_ADMIN_PAGE": "Página de administración",
 	"I18N_TOPNAV_ANDROID_APP_DESCRIPTION": "La primera aplicación de Oppia para Android ya está disponible en <strong>Inglés</strong> y <strong>Portugués brasileño</strong>. ¡Pruébala y envíanos tus comentarios!",
 	"I18N_TOPNAV_ANDROID_APP_HEADING": "Aplicación Android",
+	"I18N_TOPNAV_BLOG": "Blog",
```
We only edit `en.json` and `qqq.json` files. The other files are updated through Translatewiki.
https://github.com/oppia/oppia/wiki/Adding-new-translations-for-i18n

Please update the changes.

TejasSaraf replied:
Hey @Nik-09, I made the necessary changes, for translation i dont have translation rights to translate messages.

Nik-09 reviewed (changes requested):
Added a comment. Thanks @TejasSaraf

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

TejasSaraf committed:
Removed manual translations

Vir-8 commented on assets/i18n/kab.json:
```diff
@@ -563,6 +563,7 @@
 	"I18N_TOPIC_VIEWER_VIEW_LESS": "Sken drus",
 	"I18N_TOPNAV_ABOUT": "Ɣef",
 	"I18N_TOPNAV_ADMIN_PAGE": "Asebter n unedbal",
+	"I18N_TOPNAV_BLOG": "Ablug",
```
I think you missed to remove this change since we don't handle translation changes here as Nikhil mentioned

Vir-8 reviewed (commented):
Hi @TejasSaraf, one comment

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

TejasSaraf committed:
Removed manual translation

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

HardikGoyal2003 commented:
Hey @TejasSaraf Can you please provide a merge for this PR? If there is any problem you are facing while making progress, please feel free to discuss it. Thanks!

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

TejasSaraf committed:
Merge branch 'develop' into blogLink-22560

HardikGoyal2003 commented:
Closing this PR, as @TejasSaraf is not able to make progress on this PR.

@TejasSaraf Let me know, if there some issue at your end, I will be happy to reopen this PR again. Thanks!
